### PR TITLE
XFormInstanceSQL Deduplication

### PR DESCRIPTION
**Pull request by benrudolph**
@snopoke this allows us to do deduplication on form submission

@proteusvacuum 
:blowfish:

### CODE REVIEW

**Comment by benrudolph** [comment]
Note: first X commits are from this PR: https://github.com/dimagi/commcare-hq/pull/9018

File: corehq/form_processor/migrations/0011_add_fields_for_deprecation.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            name='orig_id',
+            field=models.CharField(max_length=255, null=True),
+            preserve_default=True,
+        ),
+        migrations.AlterField(
+            model_name='commcarecaseindexsql',
+            name='case',
+            field=models.ForeignKey(related_query_name=b'index', related_name='index_set', db_column=b'case_uuid', to_field=b'case_uuid', to='form_processor.CommCareCaseSQL'),
+            preserve_default=True,
+        ),
```
Comment (snopoke) [inline corehq/form_processor/migrations/0011_add_fields_for_deprecation.py]: oops, must have forgotten to include that in one of my changes

**Comment by snopoke** [comment]
I've started taking some notes of things to follow up on here: https://docs.google.com/document/d/1XX_WCEf8ozxt7cJoigu3R03W9vPqY45P7vFFY3stVO4/edit

Can remove / mark done as they get addressed

**Comment by benrudolph** [comment]
thanks that's helpful. will go through

**Pull request merged**